src/plugins_exts.h

```diff
@@ -214,6 +224,7 @@ struct lyplg_ext {
     lyplg_ext_data_snode_clb snode;         /**< callback to get schema node for nested YANG data */
     lyplg_ext_data_validate_clb validate;   /**< callback to validate parsed data instances according to the extension
                                                  definition */
+    lyplg_ext_get_context_clb context;      /**< callback to get extension instance context */
```

Comment:
Adding a new callback is the last resort because it changes plugins API and is useless for the great majority of extensions. But if you make sure that an extension instance is a `mount-point`, you should be able to safely access its data directly and get the context.

Reply:
The logic to retrieve the context in this case is slightly less straightforward than it sounds.  It's buried in a structure defined locally in the schema mount plugin (struct lyplg_ext_sm), so applications don't have access to that.  Perhaps that structure definition and some supporting functionality could be moved from the plugin into the library, but that sort of defeats the idea of a plugin.

Reply:
I would expect something similar as in the case of type plugins. There are lots of plugin-specific functions in the public header `plugins_types.h`. So, write a new API function `lyplg_ext_schema_mount_get_context(const struct lysc_ext_instance *ext)`, put the implementation into `schema_mount.c` and the declaration into `plugins_exts.h`. The function should then check that it is in a `mount-point` extension instance and if so, return the context. Of course, there is still the problem of which context to return (which your callback has not solved either) because there can be lots of data parsed with the context and each of those could have created a different context.

Reply:
I have implemented the new API function and removed the "internal" marking for modules.  The context returned is based on the yang-library data returned by the extension callback function.  Perhaps I'm missing something.

Reply:
Good, there is progress but your callback does not retrieve any existing context, it creates a new one every time it is called instead. I do not think that was your intention and it is most likely not the right thing to do. So you will have to solve the problem of returning a particular context somehow.